ndc-models/src/capabilities.rs

```diff
@@ -25,9 +27,12 @@ pub struct LeafCapability {}
 #[derive(Clone, Debug, PartialEq, Serialize, Deserialize, JsonSchema)]
 #[schemars(title = "Capabilities")]
 pub struct Capabilities {
-    pub query: QueryCapabilities,
-    pub mutation: MutationCapabilities,
+    pub query: Option<QueryCapabilities>,
```

Comment:
Does this make this another breaking version change? Is there any way we could avoid it, at least in the short term whilst we're still very much experimenting?

I suppose the thing we want to avoid is engine sending a non-relational query to a connector that only does relational queries, but could we express that inside `RelationalQueryCapabilities` for the time being, and then make these top-level items optional once we're committed to this course of action?

Reply:
Yeah, this is a good point, it would break things for old engines if they talk a connector that doesn't have query/mutation. Probably should revert this change and just leave it for when we stabilise this properly and then cut an NDC 0.3.0 version.